Utils/Services/ImageFetcher.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  ImageFetcher.swift
+//  Utils
+//
+//  Created by Nahuel Gladstein on 6/13/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import UIKit
+import ReactiveSwift
+
+public protocol ImageFetcherType {
+    
+    func fetchImage(_ imageURL: URL) -> SignalProducer<UIImage, ImageFetcherError>
+    
+}
+
+public enum ImageFetcherError: Error {
+    case invalidImageFormat
+    case fetchError(Error)
+}
+
+public class ImageFetcher: ImageFetcherType {
+    
+    /**
+     Fetches an image asynchronously from a URL, returning a value or an error in the SignalProducer when done.
+     - parameter imageURL: the url of the image to fetch.
+     */
+    public func fetchImage(_ imageURL: URL) -> SignalProducer<UIImage, ImageFetcherError> {
+        return URLSession.shared
+            .reactive.data(with: URLRequest(url: imageURL))
+            .flatMapError { SignalProducer(error: .fetchError($0)) }
+            .flatMap(.concat) { data, _ -> SignalProducer<UIImage, ImageFetcherError> in
```

Comment:
Depending on how is `.reactive.data(with: URLRequest(url: imageURL))` implemented. I guess it remained from ObjC that they usually had an error boolean or an error object which they compared to nil. If you can look that up. I guess that the second parameter, if it's error is useful, because if not, rac would have already deleted it from that and move it to error.